pkg/pullsync/pullsync.go

```diff
@@ -545,7 +548,13 @@ func (s *Syncer) Close() error {
 	cc := make(chan struct{})
 	go func() {
 		defer close(cc)
-		s.wg.Wait()
+		for {
```

Comment:
What is the benefit of replacing this with the wait group?

Reply:
The wait group is racy. The handler Add and this Wait causes data race. This made the tests flaky as I added the Close to the tests.
The only way to fix this would be to add another mutex. But since this waitgroup is mainly used to signal ongoing sync handlers to close, I think this is an easier approach.

Reply:
hmm, it's strange that new handle calls make it through the s.quit check